ldes-server-retention/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/retention/integrationtest/stub/InMemoryMemberPropertiesRepository.java

```diff
@@ -23,12 +23,7 @@ public InMemoryMemberPropertiesRepository() {
     }
 
     @Override
-    public void saveMemberPropertiesWithoutViews(MemberProperties memberProperties) {
-        memberPropertiesMap.put(memberProperties.getId(), memberProperties);
-    }
-
-    @Override
-    public void save(MemberProperties memberProperties) {
+    public void insert(MemberProperties memberProperties) {
         memberPropertiesMap.put(memberProperties.getId(), memberProperties);
```

Comment:
As inserting only save the memberProperties when they does not exist, I suggest the use `putIfAbsent`

Reply:
The database throws an exception if you call this twice so I adjusted the logic to throw an exception as well, when this gets called twice with the same record